### PR TITLE
Support Hibernate ORM 6.1 with Jakarta EE 10

### PR DESCRIPTION
**Pull request by papegaaij**
This addresses #406 

Migrate to Hibernate ORM 6.1 with Jakarta EE 10. Spring framework was upgraded to 6.0.0-M6, taken from their milestone repository. Support for the legacy identity generators was dropped from Hibernate.

### CODE REVIEW

**Comment by nvoxland** [comment]
Thanks. I've not gotten a chance to look super-deep at Hibernate 6 yet, so a couple questions, but mainly:

- There is no support for the old javax.persistence classes anymore right? As part of an app's upgrade from hibernate 5 to 6 they have to update their classes?

**Comment by papegaaij** [comment]
Yes, Hibernate moved to the new Jakarta namespace for Hibernate 6. There is no support for the javax annotations.

**Comment by nvoxland** [comment]
Thanks for the infor, @papegaaij 

The big trick for being able to accept this PR is how to manage support for both hibernate 5 and 6...

What we did with the similar "jakarta" package change with our servlet support is in https://github.com/liquibase/liquibase/tree/master/liquibase-core/src/main/java/liquibase/integration/servlet . We refactored as much as possible into a GenericServletListener and made the implementation-specific versions in LiquibaseJakartaServletListener vs. LiquibaseServletListener. All in the same codebase.

What we did with the hibernate 4 vs. 5 shift was to basically fork the code into a hibernate4 branch and a hibernate5 branch and release them as separate liquibase-hibernate4 vs. liquibase-hibernate5 jars. That's why our maven artifact is still "liquibase-hibernate5".

It would be nice to avoid the branch and artifact split if at all possible, and just have a single version of the code and the library. But in the case of hibernate is there a way to do that?

**Comment by papegaaij** [comment]
@nvoxland no, I don't think that will work. As you can see in the PR there were quite some incompatible API-changes, especially in the area of Dialect. Also, with all the packages renamed, the API of `EntityManager` may look the same, but as far as Java is concerned, they are completely different. Another thing is that Spring 6 requires Java 17, which is why I had to bump the Java version from 8 to 17.

I'd say, only support the Hibernate versions that are supported by Hibernate itself: 5.6 and 6.1. They plan on dropping support for 5.6 in the near future and it probably will only get some bugfixes until then. Trying to support both 5.6 and 6.1 from the same codebase is probably going to give you a lot more trouble than having to release 2 branches.

**Comment by nvoxland** [comment]
OK, thanks. Even the fact that 6.1 doesn't work with java 8 and 11 like our tests try to run gets tricky if it was all in one branch.

I'm good with doing the separate-branch trick for now. If we find it's hard to deal with we can always figure out a way to merge them later.

I'll look at the needed build logic changes quick

**Comment by nvoxland** [comment]
I don't have permissions to push to the topicusonderwijs fork apparently.

Could you take a look at the attached patch and apply it and we can see how the build goes, @papegaaij ?
[0001-Updated-build-logic-for-hibernate-6.patch.txt](https://github.com/liquibase/liquibase-hibernate/files/9757823/0001-Updated-build-logic-for-hibernate-6.patch.txt)

**Comment by papegaaij** [comment]
@nvoxland I've applied the patch

**Comment by nvoxland** [comment]
@filipelautert it's probably a good idea to remove the hiberate 3 and 4 references since those are no longer supported by hibernate

File: pom.xml [hunk 1]
```diff
@@ -67,11 +67,20 @@
...
+    </repositories>
+
+
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <hibernate.version>5.6.10.Final</hibernate.version>
-        <spring.version>5.3.22</spring.version>
-        <liquibase.version>0-SNAPSHOT</liquibase.version>
+        <hibernate.version>6.1.3.Final</hibernate.version>
+        <spring.version>6.0.0-M6</spring.version>
```
Comment (mftruso) [inline pom.xml:82]: Spring version can be bumped to the latest 6.0.2 
https://github.com/spring-projects/spring-framework/releases/tag/v6.0.2

**Comment by filipelautert** [comment]
Hello @papegaaij !  I don't have permissions to push to the topicusonderwijs fork, so could you take a look at the attached patch and apply it? It's mostly versions and documentation updates.
Liquibase version is set as 0-SNAPSHOT sothe build scripts will obtain github master head version.
[0001-Versions-and-documentation-updates.patch.txt](https://github.com/liquibase/liquibase-hibernate/files/10158339/0001-Versions-and-documentation-updates.patch.txt)

**Comment by papegaaij** [comment]
@filipelautert the patch is applied

File: pom.xml [hunk 2]
```diff
@@ -67,11 +67,21 @@
...
 
+    <repositories>
+        <repository>
+            <id>repository.spring.milestone</id>
+            <name>Spring Milestone Repository</name>
+            <url>https://repo.spring.io/milestone</url>
+        </repository>
+    </repositories>
+
+
```
Comment (DanielFran) [inline pom.xml:78]:
```suggestion
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Comment by DanielFran** [comment]
Hi @papegaaij can you apply changes? Thanks

**Review by filipelautert — APPROVED** [review]
Looks good, let's get it merged to master so more people can help testing it. Thanks for the work @papegaaij !

**Comment by filipelautert** [comment]
@papegaaij @mftruso @DanielFran it's on master now, for other fixes please open new PR's/issues. Thanks!

**Comment by steindekker** [comment]
When will liquibase-hibernate6 be available on MavenCentral?

**Comment by filipelautert** [comment]
Hi @steindekker - probably at the mid of next week. We are thinking about releasing it as hibernate6 version 4.18.0 .

**Comment by filipelautert** [comment]
It's live: https://mvnrepository.com/artifact/org.liquibase.ext/liquibase-hibernate6

**Pull request merged**